src/transformers/generation_utils.py

```diff
@@ -711,14 +800,19 @@ def greedy_search(
                 The id of the `padding` token.
             eos_token_id (:obj:`int`, `optional`):
                 The id of the `end-of-sequence` token.
+            output_attentions (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return attention weights in output
+            output_hidden_states (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return hidden states in output
+            output_scores (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return the logits in output
+
             model_kwargs:
                 Additional model specific keyword arguments will be forwarded to the :obj:`forward` function of the
                 model. If model is an encoder-decoder model the kwargs should include :obj:`encoder_outputs`.
 
         Return:
-            :obj:`torch.LongTensor` of shape :obj:`(batch_size * num_return_sequences, sequence_length)`: The generated
-            sequences. The second dimension (sequence_length) is either equal to :obj:`max_length` or shorter if all
-            batches finished early due to the :obj:`eos_token_id`.
+            :obj:`GreedySearchOutput`
```

Comment:
I think this is going to be a return type a bit more complex than that (see other comments below).

Reply:
Yes it should look more like this: https://github.com/huggingface/transformers/blob/bf713cdec7c27416f514f231ba6728cfc8135120/src/transformers/file_utils.py#L577